src/common/util.js

```diff
@@ -642,20 +642,15 @@ function replaceEndOfLineWith(text, replacement) {
   return parts;
 }
 
-function getParserName(lang, options) {
-  const supportInfo = support.getSupportInfo({ plugins: options.plugins });
-  const language = supportInfo.languages.find(
-    (language) =>
-      language.name.toLowerCase() === lang ||
-      (language.aliases && language.aliases.includes(lang)) ||
-      (language.extensions &&
-        language.extensions.some((ext) => ext === `.${lang}`))
+function inferParserByLanguage(language, options) {
+  const { languages } = getSupportInfo({ plugins: options.plugins });
+  const matched = languages.find(
+    ({ name, aliases, extensions }) =>
+      name.toLowerCase() === language ||
+      (Array.isArray(aliases) && aliases.includes(language)) ||
+      (Array.isArray(extensions) && extensions.includes(`.${language}`))
```

Comment:
This logic looks a bit wrong. It first should separately look up by language name/alias. If that yields no results, then check the extensions.

Reply:
Make sense.